classes/ActionScheduler_WPCLI_Scheduler_command.php

```diff
@@ -34,6 +37,7 @@ public function run( $args, $assoc_args ) {
 		$batch   = absint( \WP_CLI\Utils\get_flag_value( $assoc_args, 'batch-size', 100 ) );
 		$batches = absint( \WP_CLI\Utils\get_flag_value( $assoc_args, 'batches', 0 ) );
 		$clean   = absint( \WP_CLI\Utils\get_flag_value( $assoc_args, 'cleanup-batch-size', $batch ) );
+		$hooks   = explode( ',', trim( \WP_CLI\Utils\get_flag_value( $assoc_args, 'hooks', array() ) ) );
```

Comment:
🛑 Since this is inside the call to `explode()`, I believe the 3rd parameter passed to `\WP_CLI\Utils\get_flag_value()` should be an empty string instead of `array()`. Otherwise we'll get a warning like this when no `--hooks` parameter is passed:

> Warning: explode() expects parameter 2 to be string, array given in php shell code on line 1

Unfortunately, explode is a bit annoying, because if we pass an empty string, instead of returning an empty array it will return an array with only the empty string. So I think that we should possibly do something like instead:

```php
$raw_hooks = explode( ',', WP_CLI\Utils\get_flag_value( $assoc_args, 'hooks', '' ) );
$hooks     = array_filter( array_map( 'trim', $raw_hooks ) );
```
*(I'm illustrating separate variables here because the line was starting to get long)*

I think it's worth doing `array_map( 'trim' )` to ensure that no whitespace is passed in. While the shell won't pass it through if it's unquoted, it's possible someone could do this: `--hooks='foo_hook, bar_hook'`. I think it's a good idea to protect against that to reduce any annoyance to users later.

Then of course `array_filter()` will clean up any empty elements.

Reply:
Ah good catch, thanks! I'll get that fixed up tomorrow.